site/_guides/external-authorization.md

```diff
@@ -0,0 +1,413 @@
+
+```bash
+$ kubectl apply -f echo-proxy.yaml
+certificate.cert-manager.io/ingress-conformance-echo created
+httpproxy.projectcontour.io/echo created
+$ kubectl get proxies echo
+NAME   FQDN                      TLS SECRET                 STATUS   STATUS DESCRIPTION
+echo   local.projectcontour.io   ingress-conformance-echo   valid    valid HTTPProxy
+```
+
+We can verify that the application is working by requesting any path:
+
+```bash
+$ curl -k https://local.projectcontour.io/test/$((RANDOM))
+{"TestId":"","Path":"/test/12707","Host":"local.projectcontour.io","Method":"GET","Proto":"HTTP/1.1","Headers":{"Accept":["*/*"],"Content-Length":["0"],"User-Agent":["curl/7.64.1"],"X-Envoy-Expected-Rq-Timeout-Ms":["15000"],"X-Envoy-Internal":["true"],"X-Forwarded-For":["172.18.0.1"],"X-Forwarded-Proto":["https"],"X-Request-Id":["7b87d5d1-8ee8-40e3-81ac-7d74dfd4d50b"],"X-Request-Start":["t=1601596511.489"]}}
+```
+
+### Using the Authorization Server
+
+Now that we have a working application exposed by a `HTTPProxy` resource, we
+can add HTTP basic authorization by binding to the `ExtensionService` that we
+created earlier.
+The simplest configuration is to add an `authorization` field that names the
+authorization server `ExtensionService` resource that we created earlier.
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: echo
+spec:
+  virtualhost:
+    fqdn: local.projectcontour.io
+    tls:
+      secretName: ingress-conformance-echo
+    authorization:
+      extensionRef:
+        name: htpasswd
+        namespace: projectcontour-auth
+  routes:
+  - services:
+    - name: ingress-conformance-echo
+      port: 80
+
+```
+
+```bash
+$ kubectl apply -f echo-auth.yaml
+httpproxy.projectcontour.io/echo configured
+```
+
+Now, when we make the same HTTP request, we find the response requests
+authorization:
+
+```bash
+$ curl -k -I https://local.projectcontour.io/test/$((RANDOM))
+HTTP/2 401
+www-authenticate: Basic realm="default", charset="UTF-8"
+date: Fri, 02 Oct 2020 00:27:49 GMT
+server: envoy
+```
+
+Providing a user credential from the password file that we created
+earlier allows the request to succeed. Note that `contour-authserver`
+has injected a number of headers (prefixed with `Auth-`) to let the
+application know how the request has been authorized.
+
+```bash
+$ curl -k --user user1:password1 https://local.projectcontour.io/test/$((RANDOM))
+{"TestId":"","Path":"/test/27132","Host":"local.projectcontour.io","Method":"GET","Proto":"HTTP/1.1","Headers":{"Accept":["*/*"],"Auth-Handler":["htpasswd"],"Auth-Realm":["default"],"Auth-Username":["user1"],"Authorization":["Basic dXNlcjE6cGFzc3dvcmQx"],"Content-Length":["0"],"User-Agent":["curl/7.64.1"],"X-Envoy-Expected-Rq-Timeout-Ms":["15000"],"X-Envoy-Internal":["true"],"X-Forwarded-For":["172.18.0.1"],"X-Forwarded-Proto":["https"],"X-Request-Id":["2c0ae102-4cf6-400e-a38f-5f0b844364cc"],"X-Request-Start":["t=1601601826.102"]}}
+```
+
```

Comment:
Maybe a short blurb in here about how you can use anything that supports Envoy's external auth?

After you've added the Conditions support, I'd recommend some troubleshooting help as well, with some common errors and places to start.

Reply:
> Maybe a short blurb in here about how you can use anything that supports Envoy's external auth?

The second paragraph discusses that.